crates/ty_python_semantic/src/ast_node_ref.rs

```diff
@@ -87,9 +80,6 @@ where
     /// different file or Salsa revision than the module to which the node belongs.
     #[track_caller]
     pub fn node<'ast>(&self, module_ref: &'ast ParsedModuleRef) -> &'ast T {
-        #[cfg(debug_assertions)]
```

Comment:
I had to remove this assertion because the AST can now be collected so that it has a different module ref when dereferencing the node

Reply:
I think we can bring it back for the first salsa revision. It allows us to catch at least some violations in mdtests. It's always disabled in mypy primer because we use a release build